airbyte_cdk/sources/abstract_source.py

```diff
@@ -324,3 +324,16 @@ def stop_sync_on_stream_failure(self) -> bool:
         on the first error seen and emit a single error trace message for that stream.
         """
         return False
+
+    @property
+    def check_config_during_discover(self) -> bool:
+        """
+        Determines whether config validation should be skipped during discovery.
+
+        By default, config validation is not skipped during discovery. This can be overridden
+        by sources that can provide catalog information without requiring authentication.
+
+        Returns:
+            bool: True if config validation should be skipped during discovery, False otherwise.
+        """
+        return False
```

Comment:
Move this into `BaseConnector` and declare as inline class member instead. Use the same location and basic convention as `check_config_against_spec`, except that the docstring explaining the property should be in triple quotes below the member declaration instead of above as a comment.